Add documentation for Jira Helm chart

kbence opened the pull request:
It includes an updated guide and the reference of the new Helm chart.

Related to https://github.com/gravitational/teleport-private/issues/140

kbence committed:
Add documentation for Jira Helm chart

It includes an updated guide and the reference of the new Helm chart.

strideynet commented on docs/pages/kubernetes-access/helm/reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `""` | No |
+
+Jira token of the bot user to impersonate when sending the messages. Ignored when `jira.apiTokenFromSecret` is set.
```
As per https://github.com/gravitational/teleport/pull/13728#discussion_r903903467, do we want to include something here to encourage users to prefer `jira.apiTokenFromSecret` ?

kbence replied:
I've rephrased it to make it obvious that it's only intended for testing the plugin.

logand22 commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -111,6 +111,15 @@ Run `./install` from `teleport-jira` or place the executable in `/usr/bin` or `/
 </TabItem>
 </Tabs>
 
+<Details title="Installing via Helm?">
```
Should this go under the above tabs?

kbence replied:
Reformatted the document a bit

logand22 commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -136,6 +145,34 @@ By default, the Jira Teleport plugin will use a config in `/etc/teleport-jira.to
 
+<Details title="Running Teleport in Kubernetes?">
+  <Tabs>
+    <TabItem label="Self-hosted" scope={["oss","enterprise","cloud"]}>
```
Should this also have the `cloud` tab?

logand22 commented on docs/pages/enterprise/workflow/ssh-approval-jira-server.mdx:
```diff
@@ -123,6 +123,15 @@ Run `./install` from `teleport-jira` or place the executable in the appropriate
 </TabItem>
 </Tabs>
 
+<Details title="Installing via Helm?">
```
Same as above

ptgott replied:
Same comments in this guide as my comments in the Jira Cloud guide.

logand22 commented on docs/pages/enterprise/workflow/ssh-approval-jira-server.mdx:
```diff
@@ -148,6 +157,34 @@ By default, the Jira Teleport plugin will use a config in `/etc/teleport-jira.to
 
+<Details title="Running Teleport in Kubernetes?">
+  <Tabs>
+    <TabItem label="Self-hosted" scope={["oss","enterprise","cloud"]}>
```
Same as above

logand22 reviewed (approved):
Content looks good

zmb3 commented on docs/config.json:
```diff
@@ -437,6 +437,10 @@
+            },
+            {
+              "title": "teleport-plugin-jira",
+              "slug": "/kubernetes-access/helm/reference/teleport-plugin-jira/"
```
This doesn't really belong in the Kubernetes aAcess section.

Kubernetes access is about using Teleport to access Kubernetes clusters. This helm chart is about deploying Teleport's Jira plugin for access requests to a Kubernetes cluster using Helm.

@ptgott recently made some changes to move some Kubernetes things around, so let's check with him where the most appropriate place to put this would be.

kbence replied:
Yeah, good point. I've added them here to see if it makes sense, since the chart `teleport-cluster` is also more about running Teleport in Kubernetes rather than accessing a Kube cluster.

kbence replied:
I think it might be better to move all Helm charts out of this section, but I'm not sure where. @ptgott might have an idea (or a better one than moving).

ptgott replied:
This should go in the `setup/helm-reference` section

zmb3 commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -82,7 +82,7 @@ In the webhook settings page, make sure that the webhook will only send Issue Up
 We recommend installing Teleport plugins alongside the Teleport Proxy. This is an ideal
 location as plugins have a low memory footprint, and will require both public internet access
-and Teleport Auth Service access. 
+and Teleport Auth Service access.
```
Do plugins need direct access to the auth server or can they dial via proxy?

kbence replied:
They can do both. I haven't changed the docs here, but we could update this with mentioning the proxy as well

ptgott replied:
I think it's okay to limit the changes to the Helm instructions in this PR—we'll be giving the access plugin guides a top-to-bottom edit in the next sprint or two.

zmb3 commented on docs/pages/kubernetes-access/helm/reference.mdx:
```diff
@@ -18,3 +20,14 @@ Deploy the Teleport Kubernetes Service, Application Service, or Database Service
+<Tile href="./reference/teleport-plugin-jira.mdx" icon="kubernetes" title="teleport-plugin-jira">
+![Kubernetes agent](../../../img/k8s/mini-diagrams/k8s-to-teleport-mono.svg)
+
+Send notifications about acccess requests and can automatically approve them via PagerDuty.
```
The grammar here seems off.

I would either change it to "sends notifications" or remove "can".

kbence replied:
Yep, my mistake, fixing it!

zmb3 commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -136,6 +145,34 @@ By default, the Jira Teleport plugin will use a config in `/etc/teleport-jira.to
+  </Tabs>
+
+Use the following command to create a Kubernetes secret from the identity file
+generated in the [export identity](#step-37-export-the-access-plugin-identity) section:
```
Is this link correct?

kbence replied:
Oh right, it isn't. Fixing!

gozer reviewed (approved):
LGTM, apart from existing raised comments

kbence committed:
Fixed some minor mistakes in the guides

ptgott commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -111,6 +111,15 @@ Run `./install` from `teleport-jira` or place the executable in `/usr/bin` or `/
 </TabItem>
 </Tabs>
 
+<Details title="Installing via Helm?">
```
I think this would be clearest as a fourth `TabItem`. I think we should:

- Limit this `TabItem` to `helm repo add`
- Turn the instructions for starting the plugin into a `Tabs` component in the "Testing" section and add the `helm install` command there.
- Edit the "Configuration file" section to include the values file as another `TabItem`.

This way, users can have the same experience with this page for Helm as they would for other installation methods.

kbence replied:
I've changed the format a bit based on your comment, let me know if you'd change anything else there

ptgott commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -136,6 +145,34 @@ By default, the Jira Teleport plugin will use a config in `/etc/teleport-jira.to
 </TabItem>
 </Tabs>
 
+<Details title="Running Teleport in Kubernetes?">
```
Kubernetes users may not see this Details box before they create the config file mentioned above. I think we should make the Kubernetes option as visible as the option to run an executable on the host. 

I think we have two possibilities here:

1. Remove the "Self-Hosted" tab since, as far as I understand it, plugins can use the identity file to connect to the Auth Service (since they can extract the TLS credentials), but can't use the TLS credentials to connect to the Auth Service through the Proxy (because this requires SSH credentials). 

    After that, rename the "Cloud" tab to "Executable", then add a "Helm" tab. For this to make sense, we'd need to introduce the choice between deployment environments in a sentence above the `Tabs` component.
    
    We would also need to remove the "Self-Hosted" option in "Export the access-plugin certificate".

1. This hasn't been documented yet, but you can include a dropdown menu in a `Tabs` component so you can add multiple dimensions. There can be an "Executable" dropdown option with the current `TabItem`s and a `Helm` option with the `TabItem`s in the `Details` box. The current documentation is in [the comments for `Tabs`](https://github.com/gravitational/docs/blob/45d8ab1d9ecde58b0aa128e9d271281190e29878/components/Tabs/Tabs.tsx#L16-L58).

   In the second option, we would also need a sentence before the `Tabs` component introducing that there are two possible deployment environments and each has different config options for Cloud/Self-Hosted deployments.

kbence replied:
I've moved these things around to accomodate the Executable/Helm idea. What do you think? Is it easy to follow now?

ptgott commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -136,6 +145,34 @@ By default, the Jira Teleport plugin will use a config in `/etc/teleport-jira.to
+<Details title="Running Teleport in Kubernetes?">
+  <Tabs>
+    <TabItem label="Self-hosted" scope={["oss","enterprise","cloud"]}>
+
```
We need a sentence or two to put the YAML content below into context, i.e., say it's a values file and we're using it to configure the Helm chart. (This applies to both tabs.)

kbence replied:
I've added a short explanation and a link for the reference docs. What do you think?

ptgott commented on docs/pages/enterprise/workflow/ssh-approval-jira-cloud.mdx:
```diff
@@ -136,6 +145,34 @@ By default, the Jira Teleport plugin will use a config in `/etc/teleport-jira.to
+  </Tabs>
+
+Use the following command to create a Kubernetes secret from the identity file
+generated in the [export identity](#export-the-access-plugin-certificate) section:
```
```suggestion
you generated earlier:
```
I think we should prevent the reader from needing to jump around the guide.

ptgott commented on docs/pages/kubernetes-access/helm/reference.mdx:
```diff
@@ -0,0 +1,33 @@
+---
```
We're no longer keeping Helm references in the Kubernetes Access section, so we should remove this page.

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+description: Values that can be set using the teleport-plugin-jira Helm chart
+---
+
+The `teleport-plugin-jira` Helm chart is used to configure the Jira Teleport plugin which allows users to receive and manage access requests as tasks in a Jira project.
```
```suggestion
The `teleport-plugin-jira` Helm chart is used to configure the Jira Teleport plugin, which allows users to receive and manage Access Requests as tasks in a Jira project.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `""` | Yes |
+
+This parameter contains the host/port combination of the teleport auth server.
```
```suggestion
This parameter contains the host/port combination of the Teleport Auth Service or Proxy Service.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
````diff
@@ -0,0 +1,314 @@
+  auth_id: ...
+```
+
+Check out the [Jira Helm Chart documentation](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/jira#prerequisites) for more information about how to acquire these credentials!
````
1. I don't think we need an exclamation here.
2. Should this link to the new docs guide instead? Otherwise we need to maintain two sources of truth.

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `""` | Yes |
+
+Name of the Kubernetes secret that contains the credentials for the connection.
```
```suggestion
Name of the Kubernetes secret that contains the credentials for the connection to the Auth Service.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `"auth_id"` | No |
+
+Name of the key in the Kubernetes secret that holds the credentials for the connection. If the secret follows the format above, it can be omitted.
```
```suggestion
Name of the key in the Kubernetes secret that holds the credentials for the connection to the Auth Service. If the secret follows the format above, it can be omitted.
```

ptgott replied:
Also, what does this mean by "the format above"?

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `""` | Yes |
+
+Base URL of the Jira instance
```
```suggestion
Base URL of the Jira instance.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `""` | No |
+
+Jira token of the bot user to impersonate when sending the messages. Ignored when `jira.apiTokenFromSecret` is set.
```
```suggestion
Jira token of the bot user to impersonate when sending Access Request messages. Ignored when `jira.apiTokenFromSecret` is set.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `""` | Yes |
+
+Jira username / email address associated with the token.
```
```suggestion
Jira username or email address associated with the token.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `array` | `[]` | Yes |
+
+List of recipients who will receive notifications about access requests.
```
```suggestion
List of recipients who will receive notifications about Access Requests.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `""` | Yes |
+
+List of project who will receive notifications about access requests.
```
```suggestion
List of project who will receive notifications about Access Requests.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+  </TabItem>
+</Tabs>
+
+## `jira.apiTokenFromSecret`
```
I think it would help users navigate this guide more easily if we were to alphabetize the field names

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `"Task"` | Yes |
+
+List of issueType who will receive notifications about access requests.
```
Is this the caption you intended to include?

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `stdout` | No |
+
+Logger output. Could be "stdout", "stderr" or a file name, eg. "/var/log/teleport/jira.log"
```
```suggestion
Logger output. Can be `stdout`, `stderr` or a file name, eg. `/var/log/teleport/jira.log`.
```

ptgott commented on docs/pages/setup/helm-reference/teleport-plugin-jira.mdx:
```diff
@@ -0,0 +1,314 @@
+| - | - | - |
+| `string` | `stdout` | No |
+
+Logger severity. Possible values are "INFO", "ERROR", "DEBUG" or "WARN".
```
Let's use code style for the log level names

ptgott reviewed (commented):
I added some high-level comments to the Jira Cloud guide that apply to the other Access Request plugin guide changes as well.

russjones commented:
@kbence Can you make the changes requested by @zmb3 and/or @ptgott to this PR so we can merge it.

This task is not complete until documentation is available and customers can benefit from the Teleport Plugins Helm Charts.

kbence committed:
Reorganized Helm related examples

kbence committed:
Fixed a few minor mistakes

kbence committed:
Alphabetized jira.* settings

kbence committed:
Renamed Cloud to Executable in Testing

kbence committed:
Extended YAML config example with explanation

ptgott commented:
@kbence I am going to commit edits directly to this branch to help speed the process along (I will do the same with the other Helm-focused edits to our Access Request plugin guides). Please let me know if there are any issues with this!

ptgott committed:
Edit Jira Access Request plugin Helm chart docs

- Edit the configuration instructions so we don't instruct Helm users to
  use the `configure` subcommand. Also use the `Tabs` component's
  dropdown functionality so we can include both self-hosted and Teleport
  Cloud instructions for Helm and VM users.

- Minor style tweaks.

kbence committed:
Add Helm Chart example to Installation section

kbence committed:
Removed confusing (and broken) link
git push

ptgott commented:
@zmb3 Checking whether the latest version of this PR addresses your change request

kbence committed:
Fixed wrong default value

ptgott commented:
@kbence Quick question before you merge: What is the recommended way to provide a TLS key/cert to the plugin so it can run the Jira webhook on Kubernetes? The plugin doesn't include an ACME client like `teleport` does, so it won't do this on its own if you don't have TLS credentials already—this could make things difficult for Teleport Cloud customers.

While looking into refreshing the guide, I was thinking about (for the purpose of this tutorial, anyway) showing how to use Caddy to answer Let's Encrypt ACME challenges and provide TLS credentials to the plugin. While that should be pretty straightforward on a VM, I'm not sure if that's the best way to go for Kubernetes.

kbence commented:
@ptgott Yeah, my assumption was that everybody does TLS differently, so the best we can do is to provide an easy way to include it in the Helm Chart. We cloud include something generic though like `cert-manager`, I'm not sure which one is the best for our target audience. `cert-manager` also comes with a bunch of other related questions, like how to set up on AWS/GCP/DO.

I'm not sure if we could assume if someone is setting up the Jira plugin in X, they already have Teleport running in X as well, since we could then just link/include the relevant documentation for setting up `cert-manager`. What do you think?

ptgott commented:
@kbence The current guide also doesn't tell you how to provide TLS credentials to the plugin, so I'm happy to merge this PR as an improvement. The best way might be to add a Prerequisites item for TLS credentials, with (maybe) a `Details` box including some instructions for users who just want to get something set up quickly. I can look into this when I refresh the guide.
